src/components/cylc/Job.vue

```diff
@@ -94,15 +94,15 @@ const Job = (props, context) => {
 Job.props = {
   status: {
     type: String,
-    required: true
+    required: false,
```

Comment:
We're getting a very lengthy warning when this prop is `undefined` during `table.vue.spec.js`.

In the table view the status can be `undefined` here:

https://github.com/cylc/cylc-ui/blob/4f60ae6244f1ba2265cdc7454ea2bedf33450806/src/components/cylc/table/Table.vue#L67-L69

It's used to set a class on the `<rect>` element, so if it's `undefined` that's safe, at least.

The other option is to add a `latestJob` field into the mock data used by the tests

Reply:
The status field of the job, Task and Workflow icon components is required for the component to make sense (icon displays a status, don't make sense if there is no status) should never be `undefined` or `null` because we initialise these objects with state, i.e. job(status=submitted), task(status=waiting), workflow(status=stopped).

Reply:
> In fact all waiting tasks have a blank job with no latest job information.

Do you mean a blank job or no job?

It's ok for a task not to have any jobs, it's an error for a job not to have a status.

Reply:
It's not really an error, it seems reasonably useful in that it renders the space between the task icon and the task name so everything lines up.

Reply:
It's an error in the data store for a job to exist with no status.

Reply:
This isn't to do with the data store, is it? It's just that the `latestJob` property on the task object is undefined

Reply:
Right, then the table view can do `<span v-if="!latestJob" style="width: 1em" />`.

Reply:
Just tried this out, doesn't match the width though

Reply:
Most straightforward thing to do is set `visibility: hidden;` on the dummy Job component IMO

Reply:
?

https://github.com/MetRonnie/cylc-ui/pull/5